### ISSUE TITLE
PyMC-16 : exécuter réellement le GP sparse annoncé par le notebook

### ISSUE THREAD

**Issue by jsboigeEpita** [jsboige/CoursIA#13043]
## Constat vérifié sur `origin/main`

Le notebook [`PyMC-16-Sparse-Gaussian-Process.ipynb`](https://github.com/jsboige/CoursIA/blob/main/MyIA.AI.Notebooks/Probas/PyMC/PyMC-16-Sparse-Gaussian-Process.ipynb) annonce un **processus gaussien sparse** dans son nom, ses renvois et sa prose, mais son modèle exécuté est un GP latent dense :

```python
cov = pm.gp.cov.ExpQuad(input_dim=2, ls=ls)
gp = pm.gp.Latent(cov_func=cov)
f = gp.prior("f", X=X_donut)
```

Le jeu d'entraînement contient seulement 16 points. Le notebook n'instancie ni points inducteurs, ni `pm.gp.MarginalSparse`/`MarginalApprox`, ni `pm.gp.HSGP`. Ces API apparaissent uniquement dans la section markdown « Pour aller plus loin ».

L'écart est pédagogique, pas seulement nominal :

- la lecture de la longueur de corrélation affirme que « le sparse du titre désigne l'astuce computationnelle (points induits) » ;
- la bibliographie et la synthèse développent le coût `O(NM²)` et les inducing points ;
- le registre twin `probas-16-sparse-gaussian-process.yaml` déclare un « socle pédagogique commun : gaussiens parcimonieux (Sparse GP, points induits) modelisé dans les deux stacks » ;
- le jumeau Infer.NET exécute effectivement `SparseGP` avec un basis set, puis compare 4 points inducteurs à un basis contenant tous les inputs ; le côté PyMC ne réalise aucune expérience équivalente.

Le notebook démontre correctement la **classification GP dense** avec NUTS, une frontière non linéaire et une longueur d'échelle apprise. Ce contenu reste utile. Le défaut est de présenter comme exécutée la capacité sparse qui demeure seulement citée.

## Pourquoi cela compte

Un étudiant ne peut pas vérifier dans l'état actuel :

- comment choisir le nombre ou l'emplacement des points inducteurs ;
- quel compromis précision/coût introduit l'approximation sparse ;
- si HSGP ou une approximation à points inducteurs reproduit la frontière du GP dense ;
- à partir de quelle taille le passage à l'échelle devient observable.

Le cas `N=16` masque précisément la propriété distinctive annoncée : le calcul dense est trivial et ne permet aucune comparaison de coût.

## Correction attendue

Conserver le bras dense existant comme baseline, puis ajouter un **vrai bras scalable PyMC** sur un problème assez grand pour rendre la différence observable :

1. choisir une API réellement disponible dans l'environnement du notebook (`pm.gp.HSGP` ou l'API sparse/approchée PyMC appropriée) ;
2. utiliser un dataset non linéaire plus grand, tout en gardant un temps d'exécution pédagogique borné ;
3. entraîner dense et sparse/scalable sur des splits comparables ;
4. mesurer au minimum le temps, la qualité prédictive hors échantillon et l'écart des probabilités/frontières ;
5. faire varier le budget d'approximation (nombre de bases/inducteurs ou rang HSGP) pour montrer le compromis ;
6. dériver la prose uniquement des outputs réels ;
7. synchroniser honnêtement le registre twin avec la capacité effectivement exécutée.

Si aucune API sparse n'est compatible avec la classification Bernoulli retenue, le notebook doit l'établir explicitement et être renommé/requalifié en classification GP dense ; la simple mention d'API de régression sparse ne suffit pas.

## Critères d'acceptation

- [ ] Un moteur GP sparse/scalable PyMC est **instancié et exécuté**, pas seulement cité en markdown.
- [ ] Le GP dense actuel est conservé comme baseline explicite.
- [ ] Le problème comporte assez d'observations pour exercer la différence de complexité sans rendre le notebook déraisonnablement lent.
- [ ] Au moins deux budgets d'approximation sont comparés.
- [ ] Temps d'exécution et métrique prédictive hors échantillon sont reportés pour les deux bras.
- [ ] Une visualisation compare les frontières ou probabilités prédictives dense vs sparse/scalable.
- [ ] La conclusion formule honnêtement le compromis observé et ses limites.
- [ ] Le registre twin ne déclare plus de capacité sparse commune non exécutée.
- [ ] Ré-exécution complète avec outputs réels, `execution_count` cohérents et zéro erreur volontaire (C.1/C.2).

## Déduplication

Les PR historiques sur ce notebook ont porté sur sa création, ses citations, la terminologie logit/probit, les sorties et l'interprétation de la longueur d'échelle. Aucun véhicule ouvert ne touche actuellement ce chemin et aucune issue retrouvée ne demande l'exécution d'un vrai bras sparse PyMC.

**Comment by jsboigeEpita** [comment]
[CLAIMED] #13043 — myia-po-2023:CoursIA-2 2026-08-26T06:00Z — path: MyIA.AI.Notebooks/Probas/PyMC/PyMC-16-Sparse-Gaussian-Process.ipynb + registre twin probas-16-sparse-gaussian-process.yaml. Note : #13044 est un doublon exact (2 s d'écart) — la livraison référencera les deux.

**Comment by jsboigeEpita** [comment]
[DELIVERED] **PR #13077** — lane myia-po-2023:CoursIA-2, cycle c.570.

Les 9 critères d'acceptance couverts :

1. `pm.gp.HSGP(m=[15,15], c=2.5)` **instancié et exécuté** (NUTS), branché sur la likelihood `Bernoulli(logit_p=f)` existante. Le choix vs `MarginalSparse` est documenté dans le notebook : cette dernière exige une likelihood `Normal` (régression), incompatible avec la classification retenue.
2. Bras dense conservé : sections 1-5 inchangées (N=16), le Latent dense ré-exécuté sur le dataset élargi comme baseline explicite.
3. Donut élargi reproductible (seed 7, mêmes rayons) : 200 train / 100 test ; coût algébrique mesuré jusqu'à N=2000 (Cholesky 113 ms vs produit spectral 0,5 ms à N=2000).
4. Deux budgets : `m=[15,15]` (225 coeffs) et `m=[25,25]` (625).
5. Temps NUTS chronométrés par bras + accuracy et log-loss **sur le test set** : dense 69 s / 1.000 / 0.114 — HSGP-15 4 s / 1.000 / 0.113 — HSGP-25 7 s / 1.000 / 0.114.
6. Visualisation 3 panneaux (frontières dense/HSGP-15/HSGP-25 sur grille 40×40 commune + points test) + écart moyen |ΔP| = 0.006.
7. Conclusion honnête en prose post-sorties : qualité équivalente à cette échelle, l'écart est le coût (16× ici, croissant en N), limites de l'approximation (prior approximé, choix de c, domaine couvert), même idée que le basis set d'Infer-16.
8. Registre twin `probas-16-sparse-gaussian-process.yaml` : known_difference réécrit — capacité sparse **exécutée** des deux côtés (SparseGP basis set côté C#, HSGP spectral côté PyMC), attestation rebaselinée (157/157 OK).
9. Re-exécution intégrale : 16 cellules code, `execution_count` 1..16, 0 erreur, C.1 = 0.

Fix incidentel d'env : cellule §3 existante cassée sous ArviZ 1.1.0 (`az.summary()['mean'].iloc[0]` retourne désormais une str pré-formatée) — cast `float(...)` ajouté, vérifié (ls posterior 0.79).

#13044 est un doublon exact de #13043 (2 s d'écart) — à fermer comme doublon au merge.

**Comment by jsboige** [comment]
**[ai-01] Doublon exact — fermée au profit de #13044**

Cette issue et **#13044** ont un corps **byte-identique** (md5 des deux bodies confirmé) et ont été créées **à la même seconde** par le même auteur. Ce ne sont pas deux demandes voisines : c'est **une même demande enregistrée deux fois**.

La PR canonique **#13154** (PyMC-16, sur-ensemble structurel) reference #13044. La PR #13077, qui referencait celle-ci, est fermee comme doublon de livraison.

**Survivante : #13044.** Toute la suite du travail s'y rattache. Aucun contenu n'est perdu : les corps étaient identiques, et les commentaires portés ici restent lisibles sur cette page fermée.

Défaut amont tracé : **six paires** de doublons exacts sur la seule journée du 2026-08-26, toutes créées à la seconde près. Deux d'entre elles ont déjà coûté une **double livraison** (PRs #13074/#13156 sur Lean-2, #13077/#13154 sur PyMC-16) — du travail fait deux fois par des lanes qui n'avaient aucun moyen de le savoir, deux numéros distincts valant deux grains légitimes.
